tests/python/pytorch/graphbolt/impl/test_basic_feature_store.py

```diff
@@ -16,56 +16,60 @@ def test_basic_feature_store_homo():
 
     # Test read the entire feature.
     assert torch.equal(
-        feature_store.read("node", None, "a"), torch.tensor([3, 2, 1])
+        feature_store.read("node", None, "a"),
+        torch.tensor([[1, 2, 4], [2, 5, 3]]),
     )
     assert torch.equal(
-        feature_store.read("node", None, "b"), torch.tensor([2, 5, 3])
+        feature_store.read("node", None, "b"),
+        torch.tensor([[[1, 2], [3, 4]], [[2, 5], [4, 3]]]),
     )
 
     # Test read with ids.
     assert torch.equal(
-        feature_store.read("node", None, "a", torch.tensor([0, 1])),
-        torch.tensor([3, 2]),
+        feature_store.read("node", None, "a", torch.tensor([0])),
+        torch.tensor([[1, 2, 4]]),
+    )
+    assert torch.equal(
+        feature_store.read("node", None, "b", torch.tensor([0])),
+        torch.tensor([[[1, 2], [3, 4]]]),
     )
 
 
 def test_basic_feature_store_hetero():
-    a = torch.tensor([3, 2, 1])
-    b = torch.tensor([2, 5, 3])
-    c = torch.tensor([6, 8, 9])
+    a = torch.tensor([[1, 2, 4], [2, 5, 3]])
+    b = torch.tensor([[[6], [8]], [[8], [9]]])
 
     features = {}
-    features[("node", "paper", "a")] = gb.TorchBasedFeature(a)
-    features[("node", "author", "b")] = gb.TorchBasedFeature(b)
-    features[("edge", "paper:cites:paper", "c")] = gb.TorchBasedFeature(c)
+    features[("node", "author", "a")] = gb.TorchBasedFeature(a)
+    features[("edge", "paper:cites:paper", "b")] = gb.TorchBasedFeature(b)
 
     feature_store = gb.BasicFeatureStore(features)
 
     # Test read the entire feature.
     assert torch.equal(
-        feature_store.read("node", "paper", "a"), torch.tensor([3, 2, 1])
-    )
-    assert torch.equal(
-        feature_store.read("node", "author", "b"), torch.tensor([2, 5, 3])
+        feature_store.read("node", "author", "a"),
+        torch.tensor([[1, 2, 4], [2, 5, 3]]),
     )
     assert torch.equal(
-        feature_store.read("edge", "paper:cites:paper", "c"),
-        torch.tensor([6, 8, 9]),
+        feature_store.read("edge", "paper:cites:paper", "b"),
+        torch.tensor([[[6], [8]], [[8], [9]]]),
     )
 
     # Test read with ids.
     assert torch.equal(
-        feature_store.read("node", "paper", "a", torch.tensor([0, 1])),
-        torch.tensor([3, 2]),
+        feature_store.read("node", "author", "a", torch.tensor([0])),
+        torch.tensor([[1, 2, 4]]),
     )
 
 
 def test_basic_feature_store_errors():
     a = torch.tensor([3, 2, 1])
-    b = torch.tensor([2, 5, 3])
+    b = torch.tensor([[1, 2, 4], [2, 5, 3]])
 
     features = {}
-    features[("node", "paper", "a")] = gb.TorchBasedFeature(a)
+    # Test error when key does not exist.
+    with pytest.raises(AssertionError):
```

Comment:
the assertion error is thrown by dimension check or key existence check? please capture the error message as well. Reference:
```
with pytest.raises(
            RuntimeError,
            match=rf"Dataset {dataset_name} is not available.*",
        ):
```